src/librustc_lint/builtin.rs

```diff
@@ -1605,3 +1606,39 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for TrivialConstraints {
         }
     }
 }
+
+
+declare_lint! {
+    pub ELLIPSIS_INCLUSIVE_RANGE_PATTERNS,
+    Warn,
+    "`...` range patterns are deprecated"
+}
+
+
+pub struct EllipsisInclusiveRangePatterns;
+
+impl LintPass for EllipsisInclusiveRangePatterns {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(ELLIPSIS_INCLUSIVE_RANGE_PATTERNS)
+    }
+}
+
+impl EarlyLintPass for EllipsisInclusiveRangePatterns {
+    fn check_pat(&mut self, cx: &EarlyContext, pat: &ast::Pat) {
+        use self::ast::{PatKind, RangeEnd, RangeSyntax};
+
+        if let PatKind::Range(
+            _, _, Spanned { span, node: RangeEnd::Included(RangeSyntax::DotDotDot) }
+        ) = pat.node {
+            let msg = "`...` range patterns are to be deprecated";
+            let mut err = cx.struct_span_lint(ELLIPSIS_INCLUSIVE_RANGE_PATTERNS, span, msg);
+            err.span_suggestion_short_with_applicability(
+                span, "use `..=` for an inclusive range", "..=".to_owned(),
+                // FIXME: outstanding problem with precedence in ref patterns:
+                // https://github.com/rust-lang/rust/issues/51043#issuecomment-392252285
+                Applicability::MaybeIncorrect
```

Comment:
I think the `MaybeIncorrect` setting probably prevents rustfix from acting (cc @killercup, confirm?) — but otherwise it'd be nice to have some automated tests that use the suggestion (e.g., using the new `// rustfix` mode).

Reply:
Yes -- MaybeIncorrect will prevent rustfix from applying this. It will not, however, prevent the rustfix _compile tests_ as they currently exist from applying it! (Mostly because I haven't updated the test suite to also include a `// run-rustfix --yolo` mode.) So you can actually test edge cases if you want to. (Otherwise don't put them in the same file I guess.)

Reply:
OK, then @zackmdavis would you be up to add a `// run-rustfix` test for this? It's pretty straightforward: make a UI test and add that comment, and we will run rustfix and create a `foo.fixed` file. You can even use `./x.py test --bless` to create the reference file automatically.

Reply:
Monday